arrow/translate.py

```diff
@@ -0,0 +1,67 @@
+import base64
+
+
+async def translate(reader, options=None):
+    if options is None:
+        options = {}
+    t = Translator(options)
+    return t.translate(reader)
+
+
+class Translator:
+
+    def __init__(self, options=None):
+        if options is None:
+            options = {}
+        defaults = {'b64-decode-enums': False, 'prefix-object-ids': False}
+        self.options = {**defaults, **options}
+
+    def translate(self, reader):
+        enums = _list_enums(reader.writer_schema)
+        results = []
+        for record in reader:
```

Comment:
Possibly, since this is public API of `Translator`. I'm also going to experiment with the upstream code to see if it's possible to get `None` back from `fastavro.reader()`.

Reply:
`fastavro.reader()`, despite the lowercase name, is a class so `reader` here should not be none. However, that relies on code in a different file, so a `None` check does make sense. I'm not going to think too hard about what this would mean and how it could be handled. I'm just going to return `None`, which is different than the `[]` result if given a reader with no serialized entities.